language/dsl/python/src/flow.py

```diff
@@ -0,0 +1,122 @@
+
+
+from typing import Any, Dict, Optional, List
+from navigation import Navigation, NavigationFlowEndState
+from schema import Schema
+from view import View
+
+
+DataModel = Dict[Any, Any]
+
+
+class FlowResult:
+    """The data at the end of a flow"""
+    
+    def __init__(
+        self,
+        end_state: NavigationFlowEndState,
+        data: Optional[Any] = None
+    ):
+        self._end_state = end_state
+        self._data = data
+    
+    @property
+    def end_state(self) -> NavigationFlowEndState:
+        """The outcome describes _how_ the flow ended (forwards, backwards, etc)"""
+        return self._end_state
+    
+    @end_state.setter
+    def end_state(self, value: NavigationFlowEndState) -> None:
+        self._end_state = value
+    
+    @property
+    def data(self) -> Optional[Any]:
+        """The serialized data-model"""
+        return self._data
+    
+    @data.setter
+    def data(self, value: Optional[Any]) -> None:
+        self._data = value
+
+
+class Flow():
```

Comment:
Totally understand the use case for these types is tooling/generation, but we're porting types from one language to another, but in another repository too. Should we consider defining these in the Player repo? Definitely overcomplicating things, but even just looking at these I need to reference the Player repo to sanity check them

Reply:
Thats a good call out. Some of these types might need to be slightly different because they should take in DSL constructs (See https://github.com/bazel-contrib/bazelrc-preset.bzl) but largely, to your other point ideally these should be generated, so they should come from Player.